api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java

```diff
@@ -31,7 +47,87 @@
 
     public static final Date FUTURE_DATE = ProviderManagementUtils.clearTimeComponent(new Date(DATE.getTime() + 31536000000L));
 
+    @Before
+    public void init() throws Exception {
+        // execute the provider management test dataset
+        executeDataSet(XML_DATASET_PATH + XML_DATASET);
+
+        // initialize the service
+        providerManagementService = Context.getService(ProviderManagementService.class);
+    }
+    
+    @Test
+    public void testClearTimeComponent(){
+    	Date date = new Date(2018, 11, 15);
+    	date.setHours(10);
+    	date.setMinutes(32);
+    	date.setSeconds(46);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(new Date(2018, 11, 15), result);
+    }
+    
+    @Test
+    public void testClearTimeComponentofNegativeDate(){
+    	Date date = new Date(1950, -1, 2);
+    	date.setHours(15);
+    	date.setMinutes(20);
+    	date.setSeconds(12);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(new Date(1950, -1, 2), result);
+    }
+    
+    @Test
+    public void testClearTimeComponentofNewDay(){
+    	Date date = new Date(2018, 7, 20);
+    	Date result = ProviderManagementUtils.clearTimeComponent(date);
+    	Assert.assertEquals(date, result);
+    }
 
+	@Test
+   	public void shouldSetupContext() {
+   		assertNotNull(Context.getService(ProviderManagementService.class));
+   	}
+    
+    @Test
+    public void shouldAssignProviderToSupervisor() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = Context.getPersonService().getPerson(6);
+    	Person supervisor = Context.getPersonService().getPerson(2);
+    	provider.setPerson(person);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	Assert.assertFalse(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    	Assert.assertEquals(1, ProviderManagementUtils.getSupervisors(provider).size());
+    }
+    
+    @Test
+    public void supervisorListShouldBeEmptyIfNoSupervisors() throws PersonIsNotProviderException {
+    	Provider provider = new Provider();
+    	Person person = Context.getPersonService().getPerson(6);
+    	provider.setPerson(person);
+    	Assert.assertTrue(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void shouldAssignPersonToSupervisor() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
```

Comment:
It took me a second to figure out what the difference was between this one and the "shouldAssignProviderToSupervisor()", could we change the names of these two tests and make the variable names clearer?  I would always name the supervisee Person variable "supervisee" instead of "person".   And then maybe call the provider records "superviseeProvider" or "supervisorProvider"?